### DISCUSSION TITLE
Count query as a rpc function

### DISCUSSION THREAD

**Discussion by freaktree** [supabase/supabase#7831] [Questions] [answered]
Hi,

I have been trying to write this SQL QUERY into a rpc function but I cant seem to solve it. Could someone please help :)

SELECT project_symbols, COUNT(*) FILTER (WHERE project_symbols IS NOT NULL AND project_symbols NOT IN ('')) as mentions
FROM alpha_discord
GROUP BY project_symbols
ORDER BY project_symbols Desc

**Comment by steve-chavez** [comment, marked as answer]
Hello,

It seems to me that you can just wrap the query into a function. What error do you get?

```sql
create or replace function myfunc() 
returns TABLE(project_symbols alpha_discord.project_symbols%TYPE, mentions bigint) 
AS $$
  SELECT project_symbols, COUNT(*) FILTER (WHERE project_symbols IS NOT NULL AND project_symbols NOT IN ('')) as mentions
  FROM alpha_discord
  GROUP BY project_symbols
  ORDER BY project_symbols Desc
$$ language SQL;
```

**Reply by freaktree** [reply]
I'm just fairly new in using syntax and i couldn't for the life of me get the correct function setup and working. Thanks so much

My issue was i had this line set to

returns table (project_symbols text, mentions bigint) 

instead of 

returns TABLE(project_symbols alpha_discord.project_symbols%TYPE, mentions bigint)

**Reply by steve-chavez** [reply]
> project_symbols alpha_discord.project_symbols%TYPE

Note that the above is special syntax for pg to infer the type. Using the exact type should also work.

You can get the type with:
```sql
select pg_typeof(project_symbols) from alpha_discord limit 1;
```

Or of course, by looking at your table definition.
